src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp

```diff
@@ -0,0 +1,423 @@
+{
+	_retries = 4;
+
+	return write(NCP5623_LED_CURRENT, 0x00);
+}
+void
+RGBLED_NPC5623C::led_trampoline(void *arg)
+{
+	RGBLED_NPC5623C *rgbl = reinterpret_cast<RGBLED_NPC5623C *>(arg);
+
+	rgbl->led();
+}
+
+/**
+ * Main loop function
+ */
+void
+RGBLED_NPC5623C::led()
+{
+	if (!_should_run) {
+		if (_param_sub >= 0) {
+			orb_unsubscribe(_param_sub);
+		}
+
+		int led_control_sub = _led_controller.led_control_subscription();
+
+		if (led_control_sub >= 0) {
+			orb_unsubscribe(led_control_sub);
+		}
+
+		_running = false;
+		return;
+	}
+
+	if (_param_sub < 0) {
+		_param_sub = orb_subscribe(ORB_ID(parameter_update));
+	}
+
+	if (!_led_controller.is_init()) {
+		int led_control_sub = orb_subscribe(ORB_ID(led_control));
+		_led_controller.init(led_control_sub);
+	}
+
+	if (_param_sub >= 0) {
+		bool updated = false;
+		orb_check(_param_sub, &updated);
+
+		if (updated) {
+			parameter_update_s pupdate;
+			orb_copy(ORB_ID(parameter_update), _param_sub, &pupdate);
+			update_params();
+			// Immediately update to change brightness
+			send_led_rgb();
+		}
+	}
+
+	LedControlData led_control_data;
+
+	if (_led_controller.update(led_control_data) == 1) {
+		switch (led_control_data.leds[0].color) {
+		case led_control_s::COLOR_RED:
+			_r = NCP5623_LED_BRIGHT; _g = 0; _b = 0;
+			break;
+
+		case led_control_s::COLOR_GREEN:
+			_r = 0; _g = NCP5623_LED_BRIGHT; _b = 0;
+			break;
+
+		case led_control_s::COLOR_BLUE:
+			_r = 0; _g = 0; _b = NCP5623_LED_BRIGHT;
+			break;
+
+		case led_control_s::COLOR_AMBER: //make it the same as yellow
+		case led_control_s::COLOR_YELLOW:
+			_r = NCP5623_LED_BRIGHT; _g = NCP5623_LED_BRIGHT; _b = 0;
+			break;
+
+		case led_control_s::COLOR_PURPLE:
+			_r = NCP5623_LED_BRIGHT; _g = 0; _b = NCP5623_LED_BRIGHT;
+			break;
+
+		case led_control_s::COLOR_CYAN:
+			_r = 0; _g = NCP5623_LED_BRIGHT; _b = NCP5623_LED_BRIGHT;
+			break;
+
+		case led_control_s::COLOR_WHITE:
+			_r = NCP5623_LED_BRIGHT; _g = NCP5623_LED_BRIGHT; _b = NCP5623_LED_BRIGHT;
+			break;
+
+		default: // led_control_s::COLOR_OFF
+			_r = 0; _g = 0; _b = 0;
+			break;
+		}
+
+		_brightness = (float)led_control_data.leds[0].brightness / 255.f;
+		send_led_rgb();
+
+	}
+
+	/* re-queue ourselves to run again later */
+	work_queue(LPWORK, &_work, (worker_t)&RGBLED_NPC5623C::led_trampoline, this,
+		   USEC2TICK(_led_controller.maximum_update_interval()));
+}
+
+/**
+ * Send RGB PWM settings to LED driver according to current color and brightness
+ */
+int
+RGBLED_NPC5623C::send_led_rgb()
+{
+
+	uint8_t msg[7] = {0x20, 0x70, 0x40, 0x70, 0x60, 0x70, 0x80};
+	uint8_t brightness = 0x1f * _max_brightness;
+
+	msg[0] = NCP5623_LED_CURRENT | (brightness & 0x1f);
+	msg[2] = NCP5623_LED_PWM0 | (uint8_t(_r * _brightness) & 0x1f);
+	msg[4] = NCP5623_LED_PWM1 | (uint8_t(_g * _brightness) & 0x1f);
+	msg[6] = NCP5623_LED_PWM2 | (uint8_t(_b * _brightness) & 0x1f);
+
+	return transfer(&msg[0], 7, nullptr, 0);
+	return 0;
+}
+
+
+void
+RGBLED_NPC5623C::update_params()
```

Comment:
The TCA driver has. 
```
/**
 * RGB Led brightness limit
 *
 * Set to 0 to disable, 1 for minimum brightness up to 15 (max)
 *
 * @min 0
 * @max 15
 * @group System
 */
PARAM_DEFINE_INT32(LED_RGB_MAXBRT, 15);
```
This driver uses 0 - 31, but converts 0 to 1. @dagar shulde we unify this?
I wonder if this should be 0-100%